modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/filter/FDSFilterSerializerImpl.java

```diff
@@ -52,6 +52,10 @@ private void _serialize(
 			"content.Language", locale, getClass());
 
 		for (FDSFilter fdsFilter : fdsFilters) {
+			if (!fdsFilter.isEnabled()) {
```

Comment:
It looks like we can simplify this to something like
```java
if (fdsFilter.getModuleURL() == null) {
```
and remove  `isEnabled` property.

Is there a specific reason for `isEnabled` property? If this is just about simplicity vs. readability, I don't have a strong opinion, either is fine. We typically prefer simplicity in DXP.

If we are adding new property, PR is missing a semver update.

Reply:
Yeah, maybe returning `null` would be enough and won't mean a semver update...

I used an `isEnabled` method because that's what it is. It is enabled/disabled by means of an explicit flag, not by a side effect of the module URL. But if we favour the other pattern in DXP, who am I to change it? :sweat_smile: 

I'll remove it then. :+1:

Reply:
I suppose the question here is: does the user benefit from explicit API, or is this more of a sanity check? If we expect user to say "Hey, I need to disable this filter", and then needs to know he needs to overwrite URL to null to achieve it, that's bad API and we want an explicit attribute. But, if this is just a sanity check to avoid crashing if user forgets to set URL, then we don't need it.

Reply:
>  If we consider serialization as a dumb mechanism to create react props, it's up to the devs to create the right java objects and up to the FDS to interpret them.

Yeah, the problem here is that the `isEnabled` method is kind of a meta-property in the sense that it is invoked before the java filter object is read to see if that makes sense. So it's at a different level. Think of it as null vs an empty object: you can return an empty object from an object contributor, but if you want to be able to "return null" so that it is ignored you need another specific value/flag for that.

Reply:
As agreed in a private conversation I've left the `isEnabled` method and added the baseline commit because we need some way to distinguish what I've said above: the fact that a contributed filter must be ignored vs the fact that the filter's JSON object is empty.